test/test_pumactl.rb

```diff
@@ -29,16 +29,54 @@ def test_config_file
     assert_equal "t3-pid", control_cli.instance_variable_get("@pidfile")
   end
 
-  def test_environment
-    ENV.delete 'RACK_ENV' # remove from travis
+  def test_rack_env_without_environment
+    ENV.update("RACK_ENV" => "test")
     control_cli = Puma::ControlCLI.new ["halt"]
-    assert_equal "development", control_cli.instance_variable_get("@environment")
+    assert_equal "test", control_cli.instance_variable_get("@environment")
+  end
+
+  def test_environment_without_rack_env
+    ENV.delete "RACK_ENV" # remove from travis
```

Comment:
Instead of mutating the environment, which could lead to order-dependent test failures if there is another test that unexpectedly relies on a specific environment variable being set, it might be nice to reuse the [`with_env` helper](https://github.com/puma/puma/blob/860c17557c5a9d56e4d04189d637ec414ecd72f8/test/test_config.rb#L10-L22).

Or, if we are OK with adding an additional dependency, at thoughtbot we often use [`climate_control`](https://github.com/thoughtbot/climate_control) for this sort of thing. It has the advantage of being threadsafe, which might help if we plan to parallelize these tests.

Reply:
Let's just use with_env for now, but keep our eyes on climate_control for the future.